informer/src/informer_service/informer_service_handler.rs

```diff
@@ -99,3 +114,52 @@ impl InformerService {
         }
     }
 }
+
+pub struct BlockCreatedEventGenerator {
+    client: Arc<Client>,
+    rate_limiter: LeakyBucket,
+}
+
+impl BlockCreatedEventGenerator {
+    pub fn new(client: Arc<Client>) -> Self {
+        Self {
+            client,
+            rate_limiter: LeakyBucket::builder()
+                .max(100)
+                .refill_interval(Duration::from_secs(1))
+                .refill_amount(100)
+                .build()
+                .expect("There is a problem with creating rate limiter. The bucket has already been started"),
+        }
+    }
+
+    pub fn run(mut self, connection: Arc<Connection>, from_block_number: u64) -> tokio::task::JoinHandle<()> {
+        let mut current_block_number = from_block_number;
+        task::spawn(async move {
+            loop {
+                let best_block_number = self.client.best_block_header().number();
+                if best_block_number >= current_block_number {
+                    let event = self.gen(current_block_number);
+                    connection.cold_notify(&event);
+                    self.rate_limiter
+                        .acquire_one()
+                        .await
+                        .expect("Note that you are not allowed to acquire more tokens than the current");
+                    current_block_number += 1;
+                } else {
+                    tokio::time::delay_for(Duration::from_millis(500)).await;
+                }
+            }
+        })
+    }
+
+    fn gen(&mut self, favorite_block_number: u64) -> ColdEvents {
+        let current_id = BlockId::Number(favorite_block_number);
+        let block = self.client.block(&current_id).map(|block| {
+            let block = block.decode();
+            RPCBlock::from_core(block, self.client.network_id())
+        });
+        let current_block = block.expect("There is no block with the calculated Block ID");
```

Comment:
"There is no block with the calculated Block ID" is not  the reason that you can call `expect` safely.
You should give the reason why you can call `expect` safely.
You wrote an "error message". It is not a proof or reason.